hasher/src/lib.rs

```diff
@@ -80,9 +80,7 @@ pub trait Hashable: Clone {
     /// be enforced by the developer implementing the traits (see [`Hashable`] for
     ///  more details). The domain string may be parameterized by the contents of
     /// `this` and/or the generic `domain_param` argument.
-    ///
-    /// **Note:** You should always return `Some(String)`. A `None` return value
```

Comment:
This comment should remain, as returning None should not happen.  The `this` however can be None.